### ISSUE TITLE
[feature request] parent element blocking

### ISSUE THREAD

**Issue by davidhedlund** [gorhill/uBlock#2017]
### Describe the issue

Confirmed by Adblock Plus developers, it is currently not possible to also block _parent_ elements of blocked elements.

### One or more specific URLs where the issue occurs

[PornhubTV Danny Dyer Interview at SHAFTAs 2012](http://www.pornhub.com/view_video.php?viewkey=1772779562)

### Steps for anyone to reproduce the issue

The test case that the URL provide show that elements with premium video images can only be entirely blocked by blocking the parent `<li>` of `<i class="premiumIcon`.

Here's a copy of that HTML page source part:

```
<li class="videoblock videoBox" id="234648" _vkey="283773334" >
	<div class="wrap">
		<div class="phimage">
												<a href="/view_video.php?viewkey=283773334" title="Blonde Babe Rides on Couch" class="img" data-related-url="/video/ajax_related_video?vkey=283773334">
								<div class="img">
									<div class="marker-overlays">
				<var class="duration">14:50</var>
															</div>

									<img
				src="http://cdn1b.static.pornhub.phncdn.com/www-static/images/blank.gif"
				alt="Blonde Babe Rides on Couch"
				data-mediumthumb="http://i0.cdn2b.image.pornhub.phncdn.com/m=ecuK8daaaa/videos/200904/05/234648/original/12.jpg"
				data-mediabook=""
				class="js-videoThumb js-videoThumbFlip thumb"
				width="150"

							 class="rotating" id="7633648811234648"data-video-id="234648" data-prefix-id="7633648811234648" data-thumbs="16" data-path="http://i0.cdn2b.image.pornhub.phncdn.com/m=eWdT8daaaa/videos/200904/05/234648/original/{index}.jpg" data-end="7633648811234648"							title="Blonde Babe Rides on Couch" />
										</a>
									</div>
					<div class="add-to-playlist-icon display-none">
				<button type="button" data-title="Add to a Playlist" class="tooltipTrig open-playlist-link playlist-trigger" onclick="return false;" data-rel="283773334" >+</button>
			</div>
							<div class="thumbnail-info-wrapper clearfix">
				<span class="title">
																		<a href="/view_video.php?viewkey=283773334" title="Blonde Babe Rides on Couch">Blonde Babe Rides on Couch</a>
															</span>
				<span class="views"><var>131,980</var> views</span>
				<div class="rating-container up">
					<div class="main-sprite icon"></div>
					<div class="value">86%</div>
				</div>
								<var class="added">7 years ago</var>

							</div>
					</div>
</li>

			<li class="videoblock videoBox" id="89123261" _vkey="ph57d5c94f43986" >
	<div class="wrap">
		<div class="phimage">
												<a href="/view_video.php?viewkey=ph57d5c94f43986" title="Veronica Jett's pervert neighbor sneaks into her house dressed as santa" class="img" data-related-url="/video/ajax_related_video?vkey=ph57d5c94f43986">
								<div class="img">
									<div class="marker-overlays">
				<var class="duration">50:12</var>
									<span class="hd-thumbnail">HD</span>
													<i class="premiumIcon tooltipTrig" data-title="Premium Video"></i>
											</div>

									<img
				src="http://cdn1b.static.pornhub.phncdn.com/www-static/images/blank.gif"
				alt="Veronica Jett's pervert neighbor sneaks into her house dressed as santa"
				data-mediumthumb="http://i0.cdn2b.image.pornhub.phncdn.com/videos/201609/12/89123261/original/7(m=ecuK8daaaa).jpg"
				data-mediabook=""
				class="js-videoThumb js-videoThumbFlip thumb"
				width="150"

							 class="rotating" id="897485409889123261"data-video-id="89123261" data-prefix-id="897485409889123261" data-thumbs="16" data-path="http://i0.cdn2b.image.pornhub.phncdn.com/m=eWdT8daaaa/videos/201609/12/89123261/original/{index}.jpg" data-end="897485409889123261"							title="Veronica Jett's pervert neighbor sneaks into her house dressed as santa" />
										</a>
									</div>
					<div class="add-to-playlist-icon display-none">
				<button type="button" data-title="Add to a Playlist" class="tooltipTrig open-playlist-link playlist-trigger" onclick="return false;" data-rel="ph57d5c94f43986" >+</button>
			</div>
							<div class="thumbnail-info-wrapper clearfix">
				<span class="title">
																		<a href="/view_video.php?viewkey=ph57d5c94f43986" title="Veronica Jett's pervert neighbor sneaks into her house dressed as santa">Veronica Jett's pervert neighbor sneaks into her house dressed as santa</a>
															</span>
				<span class="views"><var>83</var> views</span>
				<div class="rating-container up">
					<div class="main-sprite icon"></div>
					<div class="value">100%</div>
				</div>
								<var class="added">1 week ago</var>

							</div>
					</div>
</li>
```

### Your settings

- Browser/version: Firefox 49.0
- uBlock Origin version: 1.9.6

##### Your filter lists

Default.

**Comment by davidhedlund** [comment]
[mylabel:important](https://github.com/search?utf8=%E2%9C%93&q=commenter%3Adavidhedlund+%22mylabel%3Aimportant%22+in%3Acomments&type=Issues&ref=searchresults)

**Comment by gorhill** [comment]
Closing and marking as _invalid_, as per [CONTRIBUTING](https://github.com/gorhill/uBlock/blob/master/CONTRIBUTING.md).

**Comment by gwarser** [comment]
@gorhill `pornhub.com###relateRecommendedItems .videoblock:has(.premiumIcon)` i can say "work" in Chromium, but in Firefox element is restyled dynamically and visible after a moment.

**Comment by gorhill** [comment]
Duplicate of #1912.
